### PR TITLE
WIP Infrastructure - openshift-registry-proxy pod

### PR DESCRIPTION
**Pull request by jwhonce**
* expose openshift registry to default IP

### CODE REVIEW

**Comment by jwhonce** [comment]
@danmcp Please review. This is the pod that allows Steve's students to push containers into the docker registry deployed in the cluster from their host OS.  Once this is merged, I'll update the jenkins jobs.

**Comment by danmcp** [comment]
@jwhonce I am fine with the impl.  @smarterclayton Any objection to this logic sitting in origin?

File: hack/build-images.sh [hunk 1]
```diff
@@ -44,9 +44,10 @@ cp -f "${imagedir}/dockerregistry" images/dockerregistry/bin
 
 # builds an image and tags it two ways - with latest, and with the release tag
 function image {
+  set -x
   echo "--- $1 ---"
   docker build -t $1:latest $2
-  docker tag $1:latest $1:${OS_RELEASE_COMMIT}
+  docker tag -f $1:latest $1:${OS_RELEASE_COMMIT}
```
Comment (smarterclayton) [inline hack/build-images.sh]: Why?

**Comment by smarterclayton** [comment]
This looks like this is going to break as soon as we turn on auth security, because the registry cert will not be valid.  Why is this necessary again?

**Comment by smarterclayton** [comment]
Why can't they push directly to the registry?  The service proxy should be available on the host...

File: hack/build-images.sh (hunk 1 above)
Comment (jwhonce) [inline hack/build-images.sh]: @smarterclayton  While iterating during testing, I got a set of errors this fixed.  I can pull it now.

**Comment by jwhonce** [comment]
@smarterclayton This proxy allows the host OS to see the registry without having to edit /etc/hosts or any DNS hacks.  For Steve's windows and mac users that was proving to be difficult.

**Comment by smarterclayton** [comment]
I would keep the image outside of Openshift and set up a docker trusted build for it.  And then we can just leave the pod json here.  I don't really want to slow down our builds for something that is basically replicating what the router does but in a less efficient way.  

Also, why can't this use the router?  Or do Steves labs not include that?

**Comment by jwhonce** [comment]
@smarterclayton When I spoke with others no one had a way to use the router to expose the registry on the VM main IP.  Do you?  I would much rather do that than carry this pod.

**Comment by smarterclayton** [comment]
If you have the router IP exposed to the host you should only need to create a route in the default namespace pointing to the registry (to push form the master to the hosts).  Otherwise, I'm not aware of anything that would block it.

**Pull request closed without merging**